code/datums/flock/flock.dm

```diff
@@ -541,33 +540,35 @@
 		if (RL_Started) RL_UPDATE_LIGHT(T)
 
 	for(var/obj/O in T)
-		if(istype(O, /obj/machinery/door/feather))
-			// repair door
-			var/obj/machinery/door/feather/door = O
-			door.heal_damage()
-			animate_flock_convert_complete(O)
 		for(var/keyPath in flock_conversion_paths)
-			var/obj/replacementPath = flock_conversion_paths[keyPath]
 			if(istype(O, keyPath))
-				if(isnull(replacementPath))
-					qdel(O)
-				else
-					var/dir = O.dir
-					var/obj/converted = new replacementPath(T)
-					// if the object is a closet, it might not have spawned its contents yet
-					// so force it to do that first
-					if(istype(O, /obj/storage))
-						var/obj/storage/S = O
-						if(!isnull(S.spawn_contents))
-							S.make_my_stuff()
-					// if the object has contents, move them over!!
-					for (var/obj/OO in O)
-						OO.set_loc(converted)
-					for (var/mob/M in O)
-						M.set_loc(converted)
-					qdel(O)
-					converted.set_dir(dir)
-					animate_flock_convert_complete(converted)
+				if (istype(O, /obj/machinery/door))
+					if (istype(O, /obj/machinery/door/firedoor/pyro))
+						qdel(O)
+						continue
+					if (istype(O, /obj/machinery/door/window) || istype(O, /obj/machinery/door/airlock/pyro/glass/windoor) || istype(O, /obj/machinery/door/unpowered/wood))
+						qdel(O)
+						continue
+					if (istype(O, /obj/machinery/door/poddoor/pyro) && !(findtext(O.name, "cargo") || findtext(O.name, "pod") || findtext(O.name, "mass") || findtext(O.name, "disposal"))) // shutters
```

Comment:
This cannot be the best way of doing this. Is there a particular reason we can't just convert them all?

Reply:
Windoors are found over tables, and for the last case, some of these doors are found in glass windows, such as for the ai core shield, or over tables, such as for the bridge lockdown shutters, meaning if they were converted there would be tables inside doors.

Didn't seem like sliding windoors and sliding windows were big enough for Flock door creation too in terms of making sense.

Reply:
Would work, but seemed like some shutters should be converted, whereas others shouldn't, like chemistry shutters felt like they shouldn't be converted, where ones directly connected to space should. Idk.